iteration/iteration.go

```diff
@@ -0,0 +1,88 @@
+package iteration
+
+import (
+	"context"
+	"time"
+
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormsupport"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	uuid "github.com/satori/go.uuid"
+)
+
+// Iteration describes a single iteration
+type Iteration struct {
+	gormsupport.Lifecycle
+	ID        uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
+	ProjectID uuid.UUID `sql:"type:uuid"`
+	ParentID  uuid.UUID `sql:"type:uuid"` // TODO: This should be * to support nil ?
+	StartAt   *time.Time
```

Comment:
After some discussion, I think StartAt and EndAt make the most sense.

* StartsAt is future tense which makes a historical Iteration named oddly
* StartedAt is past tense which makes an upcoming Iteration named oddly